crates/red_knot_python_semantic/resources/mdtest/call/annotation.md

```diff
@@ -0,0 +1,43 @@
+# `typing.Callable`
+
+```py
+from typing import Callable
+
+def _(c: Callable[[], int]):
+    reveal_type(c())  # revealed: int
+
+def _(c: Callable[[int, str], int]):
+    reveal_type(c(1, "a"))  # revealed: int
+
+    # error: [invalid-argument-type] "Object of type `Literal["a"]` cannot be assigned to parameter 1; expected type `int`"
+    # error: [invalid-argument-type] "Object of type `Literal[1]` cannot be assigned to parameter 2; expected type `str`"
+    reveal_type(c("a", 1))  # revealed: int
+```
+
+The `Callable` annotation can only be used to describe positional-only parameters.
+
+```py
+def _(c: Callable[[int, str], None]):
+    # error: [unknown-argument] "Argument `a` does not match any known parameter"
+    # error: [unknown-argument] "Argument `b` does not match any known parameter"
+    # error: [missing-argument] "No arguments provided for required parameters 1, 2"
+    reveal_type(c(a=1, b="b"))  # revealed: None
```

Comment:
there's room for the error message to be improved here, but it's out of scope for this PR:

```suggestion
def _(c: Callable[[int, str], None]):
    # error: [unknown-argument] "Keyword argument `a` does not match any known parameter (`c` accepts no keyword arguments)"
    # error: [unknown-argument] "Keyword argument `b` does not match any known parameter (`c` accepts no keyword arguments)"
    # error: [missing-argument] "No arguments provided for required parameters 1, 2"
    reveal_type(c(a=1, b="b"))  # revealed: None
```

Reply:
Yeah, totally.

Reply:
#16919, feel free to update the description or comment to provide more context and examples.